### PR TITLE
Updates Create Attribute example.

### PR DESCRIPTION
**Pull request by tcr3dr**
This addresses some of the questions brought up by #277 and #420.

The goal is "the correct way for users to implement custom attributes". I rewrote this example to leverage the vehicle_class property of connect to instantiate a custom Vehicle class. The Vehicle constructor works similarly to how the internal ones, allowing you to add private properties, message listeners, etc. that become part of the class definition.

I recommend this as the definitive way downstream users implement custom properties, as it's similar to how internal extensions like [dronekit-python-solo](https://github.com/dronekit/dronekit-python-solo) do it.

### CODE REVIEW

**Comment by tcr3dr** [comment]
A key aspect of this approach is that we fully utilize Python's class system, allowing documentation and tooling to react normally. With multiple inheritance, you can support multiple Vehicle "extensions", though I suspect there's no need for more than one custom Vehicle class in a given application.

File: examples/create_attribute/create_attribute.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 Full documentation is provided at http://python.dronekit.io/examples/create_attribute.html
 """
 
-from dronekit import connect
+from dronekit import connect, Vehicle
```
Comment (hamishwillee) [inline examples/create_attribute/create_attribute.py]: Do we really need a model where we separately import from dronekit and dronekit.lib? Would be cool to not have to think about where you import from.

File: examples/create_attribute/create_attribute.py [hunk 2]
```diff
@@ -69,48 +65,56 @@ def __str__(self):
...
+        @self.on_message('RAW_IMU')
+        def listener(self, name, message):
+            """
+            The listener is called for messages that contain the string specified in the decorator,
+            passing the vehicle, message name, and the message.
+            
+            The listener writes the message to the (newly attached) ``vehicle.raw_imu`` object 
+            and notifies observers.
+            """
+            self.raw_imu.time_boot_us=message.time_usec
```
Comment (hamishwillee) [inline examples/create_attribute/create_attribute.py]: These should all be ``self._raw_imu`` right? (i.e. with the private member)

File: examples/create_attribute/create_attribute.py (hunk 1 above)
Comment (tcr3dr) [inline examples/create_attribute/create_attribute.py]: Oh, this should change only to `from dronekit import ____`. Didn't notice the `.lib` part.

File: examples/create_attribute/create_attribute.py (hunk 2 above)
Comment (tcr3dr) [inline examples/create_attribute/create_attribute.py]: Yes.

**Comment by hamishwillee** [comment]
OK, looks really great. I think people are going to want to potentially create their own libraries for re-usability. Can you split into two files if you agree? 

Bonus points if we can accept files of the right format as a "plugin" to our libraries - ie so that we can take an external file of the right format and dump it into our source tree and have everything work automatically :-) 

Drop me a line when you're done (whatever you decide should be done) and I'll look at a docs update.

**Comment by tcr3dr** [comment]
@hamishwillee 

> Can you split into two files if you agree?

To clarify, separate the Vehicle class definition from the example? I can do that.

**Comment by hamishwillee** [comment]
>> Can you split into two files if you agree?

> To clarify, separate the Vehicle class definition from the example? I can do that.

Yes please, and fix up the other minor bugs raised above.

**Comment by tcr3dr** [comment]
@hamishwillee Bumped.

**Comment by hamishwillee** [comment]
@tcr3dr Thanks for making the fixes. I think this is a really good example now - both clear and useful. I've added docs for it, and also removed a couple of unnecessary imports in both files. Should be ready to merge.

**Pull request merged**